public/js/controllers/index.js

```diff
@@ -57,10 +57,33 @@ angular.module('mean.system')
         };
 
         $scope.playAsGuest = () => {
-          game.joinGame();
-          $location.path('/app');
+          if ($scope.region === undefined || $scope.region === '') {
+            $scope.error = 'Select a region first!';
+          } else {
+            $('#close').click();
+            $location.path('/app');
+          }
+        };
+
+        $scope.playWithStrangers = () => {
+          if ($scope.region === undefined || $scope.region === '') {
+            $scope.error = 'Select a region first!';
+          } else {
+            $('#close').click();
+            $location.path('/app');
+          }
         };
 
+        $scope.playWithFriends = () => {
+          if ($scope.region === undefined || $scope.region === '') {
+            $scope.error = 'Select a region first!';
+          } else {
+            $('#close').click();
+            $location.path('/app');
+          }
+        };
+
+
```

Comment:
Good Job!

Just a quick suggestion, since you seem to be repeating your self when you check if a card has been played, why not extract it into a function and call it when you need to

Reply:
Nice suggestion Sam, but they aren't totally doing the same thing, extracting to function will affect the route. Thanks